### DISCUSSION TITLE
High CPU & Ram usage on specific page.

### DISCUSSION THREAD

**Discussion by jinmvp** [cefsharp/CefSharp#4354] [General]
Hi, recently i've looking for the solution for my app high CPU usage. I ended up with the conclusion that the URL web page is the main reason and im using WinForms with CefSharp.

What version of the product are you using?

Nuget Version : 105.3.330

What architecture x86 or x64?: x86
On what operating system? : Win 10 x64
Are you using WinForms, WPF or OffScreen? : WinForms

What steps will reproduce the problem?
The URL i have problem with: 
[HIGH CPU URL](https://www.walmart.com/blocked?url=L2FjY291bnQvc2lnbmluL3dpdGhvdHBjaG9pY2U/dmlkPW9hb2gmdGlkPTAmcmV0dXJuVXJsPSUyRg==&uuid=7b301260-89b4-11ed-a964-072a4a8173f7&vid=76d14eb7-89b4-11ed-8057-5a50524d4f54&g=b)

This will cause high CPU & RAM usage when navigate to this URL.

My CefSharp command lines im using:

`settings.CefCommandLineArgs.Add("disable-image-loading", "1");`
`settings.CefCommandLineArgs.Add("--disable-notifications");`
`settings.CefCommandLineArgs.Add("--use-gl=swiftshader");`
`settings.CefCommandLineArgs.Add("--disable-gpu");`

What is the expected output? What do you see instead?
Expected LOW CPU usage. I only see high cpu usage.

What i've tried:
https://github.com/cefsharp/CefSharp/discussions/3892

**Comment by amaitland** [comment]
Converted to [Discussion](https://github.com/cefsharp/CefSharp/discussions) as this is unlikely a bug in `CefSharp`, bug report was also incomplete. A new issue can be opened if code changes are required. 

There are quite a number of pieces of the [Bug Report Template](https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report) you skipped over. 

> Please only create an issue if you can reproduce the problem with version 107.1.90 or greater.

As per the [Bug Report Template](https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report). 

Have you tested with the current supported version? `M108` is the current version.

> Does the problem reproduce using the [MinimalExample](https://github.com/cefsharp/CefSharp.MinimalExample)?

Did you test with the `MimimalExample`?

> My CefSharp command lines im using:

`GPU Acceleration` should be left enabled under normal circumstances, `Swiftshader` is no longer supported by `Chromium`, setting a specific `--use-gl` is not recommended.

Make sure your application has an [app.manifest(https://github.com/cefsharp/CefSharp/wiki/Quick-Start-For-MS-.Net-4.x#40-add-appmanifest-to-your-application) with the compatibility entries so `GPU detection` works correctly.

Test with the `MinimalExample` as it has an app.manifest

> Expected LOW CPU usage. I only see high cpu usage.

How does `Chrome` behave?

> What i've tried:
> #3892

Tried what exactly? Just linking to a discussion doesn't tell me exactly what **you** have tried.

**Reply by jinmvp** [reply]
I still got the problem with 108.4.130.

I also tested with MimimalExample and got the same problem.

I tested again with only 

` settings.CefCommandLineArgs.Add("disable-image-loading", "1");`

I tried to deleted 3 files 

- 'vk_swiftshader.dll'

- 'vk_swiftshader_icd.json'

- 'vulkan-1.dll'
My app has app.manifest file with
`            <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/>
`

**Reply by amaitland** [reply]
> I also tested with MimimalExample and got the same problem.

You'll need to test with the `CEF Sample Application(cefclient)` then. Details at https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef

**Reply by jinmvp** [reply]
I found the problem, turned out its the page im trying to load using too much JS and taken alot of CPU resources to calculated.

**Reply by amaitland** [reply]
Does the problem reproduce using the `CEF Sample Application(cefclient)`?

How does it behave in `Chrome`?

**Reply by jinmvp** [reply]
Yes  the problem is still there with Chrome and CEF Sample Application(cefclient), the URL using JS to calculated so it using alot of CPU resources.
